components/DisplayAddress/DisplayAddress.tsx

```diff
@@ -14,16 +14,17 @@ export function DisplayAddress({
 }: DisplayAddressProps) {
   const { jsonRpcProvider } = useConfig();
   const [gotResponse, setGotResponse] = useState(false);
-  const [addr, setAddr] = useState<string>(address);
+  const [name, setName] = useState<string | null>(null);
```

Comment:
There's no need to store the address in state, since we have it from props. We only use the state for the data we fetch. If, after fetching, we didn't get an ENS name we just fall back to the address from props.